### PR TITLE
fix(bounty): allow manual numeric entry

### PR DESCRIPTION
**Pull request by LIghtJUNction**
# LMM API Pull Request

> LMM API 是 New API 的衍生版本。请说明本次变更与上游的关系，并确保 PR 只包含当前任务所需的改动。

## 变更说明 / Summary

修复创建悬赏时金额与名额无法稳定手动填写的问题。数值输入在编辑过程中保留文本状态，校验及提交时再转换为数字；金额输入允许合法小数，名额仍限制为 1–100 的整数。补充回归测试，覆盖手动输入数字文本和空值校验。

本变更由 AI 辅助完成，并由仓库维护者执行验证、签名提交和发布。

## 与上游的关系 / Upstream relationship

- [x] LMM API 独有变更 / LMM API-specific change
- [ ] 上游尚未合并的修复或功能 / Change not yet merged upstream
- [ ] 同步或移植上游变更 / Upstream sync or backport
- [ ] 不适用（文档、CI、仓库维护等）/ Not applicable

上游参考 / Upstream reference: N/A，开源悬赏是 LMM API 独有功能。

## 关联 Issue / Related issue

Closes #

## 变更类型 / Type of change

- [x] Bug 修复 / Bug fix
- [ ] 新功能 / Feature
- [ ] 重构或性能优化 / Refactor or performance
- [x] 前端或交互 / Frontend or UX
- [ ] 部署、CI 或构建 / Deployment, CI, or build
- [ ] 文档或仓库维护 / Documentation or maintenance

## 验证 / Verification

```text
command: bun test src/features/open-source-bounties/__tests__/validation.test.ts
result: 8 pass, 0 fail

command: bun run typecheck
result: passed

command: bun run format:check
result: passed

command: bunx oxlint -c .oxlintrc.json src/features/open-source-bounties/index.tsx src/features/open-source-bounties/validation.ts src/features/open-source-bounties/__tests__/validation.test.ts
result: passed
```

## 截图或日志 / Screenshots or logs

N/A；回归由输入状态与校验测试覆盖。

## 提交检查 / Checklist

- [x] 我已搜索本仓库的 Issues 和 PRs，确认没有重复工作。
- [x] 此 PR 只包含与当前目标相关的改动，未混入格式化或其他无关变更。
- [x] 我已说明该变更与上游 New API 的关系，并保留必要的来源、版权和许可证信息。
- [x] 我已检查兼容性；如有破坏性变化，已在说明中明确列出迁移方式。
- [x] 我已补充或更新相关测试，并在上方记录实际验证结果；如未测试，已说明原因。
- [x] 我已更新受影响的文档、示例配置和多语言文案（如适用）。
- [x] 代码、日志、截图和提交记录中不包含 API Key、Cookie、Token、密码、DSN 或其他敏感信息。

## Summary by Sourcery

允许赏金奖励金额和名额数量以文本形式输入和存储，将数值转换延后到校验和提交阶段，以改善手动输入的处理体验。

Bug Fixes:
- 修复在手动输入奖励金额和名额值时的稳定性问题，支持数值文本输入，包括金额的小数，同时对名额强制限制为 1–100 的整数。
- 确保空的奖励金额和名额输入保持为无效状态，并提供清晰的校验错误信息。

Enhancements:
- 调整草稿费用计算和草稿保存逻辑，仅在计算和提交时将基于文本的奖励字段转换为数字。
- 更新表单字段配置，以支持奖励金额的任意小数步长，同时保持名额输入限制为整数。

Tests:
- 添加回归测试，覆盖奖励字段的手动数值文本输入，以及对空输入的校验行为。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Allow bounty reward amount and slot count to be entered and stored as text, with numeric conversion deferred to validation and submission to improve manual input handling.

Bug Fixes:
- Fix instability when manually entering reward amount and slot values by accepting numeric text input, including decimals for amounts and enforcing 1–100 integers for slots.
- Ensure empty reward amount and slot inputs remain invalid and surface clear validation errors.

Enhancements:
- Adjust draft charge calculation and draft saving logic to convert text-based reward fields to numbers only at computation and submission time.
- Update form field configuration to support arbitrary decimal steps for reward amounts while keeping slot input constrained to whole numbers.

Tests:
- Add regression tests covering manual numeric text entry for reward fields and validation behavior for empty inputs.

</details>

### CODE REVIEW

File: apps/web/src/features/open-source-bounties/__tests__/validation.test.ts [hunk 1]
```diff
@@ -42,6 +42,31 @@ describe('bounty draft validation', () => {
     assert.deepEqual(validateBountyDraft(VALID_DRAFT), {})
   })
 
+  test('accepts manually entered numeric text while keeping empty inputs invalid', () => {
```
[the remark this answers is not in the document]
Comment (LIghtJUNction) [reply apps/web/src/features/open-source-bounties/__tests__/validation.test.ts:46]: Verified in a86309a20: validation now covers whitespace-only, malformed, and trimmed numeric strings, and the shared parser also protects preview/save conversion. `bun test src/features/open-source-bounties/__tests__/validation.test.ts` passes 9/9; typecheck, format, lint, and the full PR checks are green.
